hms-app/src/components/card-details/VotingHandler.tsx

```diff
@@ -0,0 +1,206 @@
+            message: undefined,
+            icon: <Check />,
+            autoClose: 2000,
+          })
+        }
+      }
+    )
+  }
+
+  const removeThisVote = () => {
+    removeVote(removeIdeaVoteParticipant, setVoteCheck)
+  }
+
+  const addVote = async (
+    action = createIdeaVoteParticipant,
+    check = setVoteCheck
+  ) => {
+    if (participantInfo.participantId === '') {
+      showNotification({
+        id: 'participant-load',
+        color: 'red',
+        title: 'Not participating in hackathon',
+        message: 'You must join the hackathon first to vote.',
+        icon: <X />,
+        autoClose: 5000,
+      })
+      return
+    }
+    setButtonisDisabled(true)
+    showNotification({
+      id: 'participant-load',
+      loading: true,
+      title: `Adding vote to: "${ideaData.title}"`,
+      message: undefined,
+      autoClose: false,
+      disallowClose: false,
+    })
+    action(instance, ideaData.id, participantInfo.participantId).then(
+      (response) => {
+        setButtonisDisabled(false)
+        setLoader(true)
+        if (JSON.stringify(response).toString().includes('error')) {
+          check(false)
+          updateNotification({
+            id: 'participant-load',
+            color: 'red',
+            title: 'Failed to add vote',
+            message: undefined,
+            icon: <X />,
+            autoClose: 2000,
+          })
+        } else {
+          check(true)
+          updateNotification({
+            id: 'participant-load',
+            color: 'teal',
+            title: `Vote added to: "${ideaData.title}"`,
+            message: undefined,
+            icon: <Check />,
+            autoClose: 2000,
+          })
+        }
+      }
+    )
+
+    useEffect(() => {
+      if (findVoter()) setVoteCheck(!!findVoter())
+    }, [ideaData])
+
+    const loadIdeaData = () => {
+      getIdeaDetails(instance, ideaData.id).then((data) => {
+        setIdeaData(data)
+        setLoader(false)
+      })
+    }
+
+    useEffect(() => {
+      loadIdeaData()
+    }, [loader])
+  }
+
+  const addThisVote = async () => {
+    await addVote(createIdeaVoteParticipant, setVoteCheck)
+  }
+
+  useEffect(() => {
+    if (user) {
+      setParticipantInfo({
+        userId: user.id,
+        participantId: hackathonParticipantId,
+      })
+    }
+  }, [user, hackathonParticipantId])
+
+  return (
+    <Card.Section className={classes.noBorderSection}>
+      <Stack align={'center'} spacing={'xs'}>
+        <Text className={classes.label}>Votes: </Text>
+        <Text className={classes.text}>{ideaData.voters?.length}</Text>
+      </Stack>
+
+      <Button
+        disabled={buttonIsDisabled}
+        onClick={voteCheck ? removeThisVote : addThisVote}
+        style={{
+          backgroundColor: voteCheck ? LEAVE_BUTTON_COLOR : JOIN_BUTTON_COLOR,
+        }}
+      >
+        {voteCheck ? 'Remove Vote' : 'Add Vote'}
+      </Button>
+    </Card.Section>
```

Comment:
**BUG**
- Counter not updating automatically after adding vote
- Button stays blue after voting and refreshing page. Turns red only after clicking